### PR TITLE
Bug fix #1390

### LINKED ISSUE #1390: API Error: PIREP fields array to string conversion
Needs validations

```
POST /api/pireps/pirepid_1/fields
{
	"fields": {
		"fields": {
			"key": "value"
		}
	}
}
```

Returns:

```
{
  "type": "https://phpvms.net/errors/internal-error",
  "title": "Array to string conversion",
  "details": "Array to string conversion",
  "status": 503,
  "error": {
    "status": 503,
    "message": "Array to string conversion"
  },
  "original_exception": "ErrorException",
  "trace": {
    "file": "/var/www/demo.phpvms/vendor/laravel/framework/src/Illuminate/Support/Str.php",
    "line": 596,
    "function": "handleError",
    "class": "Illuminate\\Foundation\\Bootstrap\\HandleExceptions",
    "type": "->",
    "args": [
      8,
      "Array to string conversion",
      "/var/www/demo.phpvms/vendor/laravel/framework/src/Illuminate/Support/Str.php",
      596,
      {
        "search": "?",
        "replace": [
          1,
          "2022-01-24 22:06:52",
          "2022-01-24 22:06:52"
        ],
        "subject": "insert into `pirep_field_values` (`pirep_id`, `name`, `slug`, `value`, `source`, `updated_at`, `created_at`) values (?, ?, ?, ?, ?, ?, ?)",
        "segments": [
          ", ",
          ", ",
          ", ",
          ", ",
          ", ",
          ", ",
          ")"
        ],
        "result": "insert into `pirep_field_values` (`pirep_id`, `name`, `slug`, `value`, `source`, `updated_at`, `created_at`) values (pirepid_1, fields, fields, ",
        "segment": ", "
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by YashGovekar**
Fixes #1390 
Compared request array with model fillables.

### CODE REVIEW

File: app/Http/Controllers/Api/PirepController.php [hunk 1]
```diff
@@ -116,8 +116,18 @@ protected function getFields(Request $request): ?array
...
-        foreach ($request->input('fields') as $field_name => $field_value) {
+        $fillable = (new PirepFieldValue())->getFillable();
+        if (! empty(array_diff($fillable, $request_fields))) {
+            if (array_key_exists('fields', $request_fields)) {
+                $request_fields = $request->input('fields')['fields'];
+            } else {
+                return [];
+            }
+        }
+        foreach ($request_fields as $field_name => $field_value) {
```
Comment (nabeelio) [inline app/Http/Controllers/Api/PirepController.php:130]: I'm not sure what you're doing here, the model already enforces fillable fields?

Comment (YashGovekar) [reply app/Http/Controllers/Api/PirepController.php:130]: Yes, but I have added validation as per the given object in your issue:

```
"fields": {
		"fields": {
			"key": "value"
		}
	}

**Comment by nabeelio** [comment]
This was for something different, I have notes for it somewhere. It might be closed now anyway

**Comment by YashGovekar** [comment]
> This was for something different, I have notes for it somewhere. It might be closed now anyway

Ah Okay

**Pull request closed without merging**